cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/upgradetemplates/UpgradeTemplatesDialog.java

```diff
@@ -0,0 +1,149 @@
+package com.devonfw.cobigen.eclipse.upgradetemplates;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+import com.devonfw.cobigen.api.util.CobiGenPaths;
+import com.devonfw.cobigen.api.util.TemplatesJarUtil;
+import com.devonfw.cobigen.eclipse.common.constants.InfrastructureConstants;
+import com.devonfw.cobigen.eclipse.common.constants.external.CobiGenDialogConstants.UpgradeTemplatesDialogs;
+import com.devonfw.cobigen.eclipse.common.exceptions.GeneratorProjectNotExistentException;
+import com.devonfw.cobigen.eclipse.common.tools.ResourcesPluginUtil;
+import com.devonfw.cobigen.impl.CobiGenFactory;
+
+/**
+ * Dialog to upgrade old monolithic templates
+ */
+public class UpgradeTemplatesDialog extends Dialog {
+
+  /** Logger instance. */
+  private static final Logger LOG = LoggerFactory.getLogger(UpgradeTemplatesDialog.class);
+
+  /**
+   * Creates a new {@link UpgradeTemplatesDialog}
+   */
+  public UpgradeTemplatesDialog() {
+
+    super(Display.getDefault().getActiveShell());
+
+  }
+
+  @Override
+  protected Control createDialogArea(Composite parent) {
```

Comment:
I'm not sure about this.
The HealthCheck already uses: openErrorDialogWithContextUpgrade. Maybe do it similar like this instead.